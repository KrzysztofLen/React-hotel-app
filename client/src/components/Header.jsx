import React from 'react';
import LoginSystem from './LoginSystem/LoginSystem';
import {Logo} from './Logo/Logo';
import {HotelsFilterContainer} from "./Search/Search";
import {getFilteredHotels} from "../selectors/getFilteredHotels";
import {connect} from "react-redux";

export class Header extends React.Component {

	render() {
		return (
			<header className="header">
				<Logo/>
				{/*<Search/>*/}
				<HotelsFilterContainer/>
				<LoginSystem/>
			</header>
		)
	}
}

const mapStateToProps = state => {
	return {
		hotels: getFilteredHotels(state.hotels, state.hotelsSearch)
	};
};

export default connect(mapStateToProps, null)(Header);
