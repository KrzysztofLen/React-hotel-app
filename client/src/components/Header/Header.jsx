import React, {Component} from 'react';
import LoginSystem from '../LoginSystem/LoginSystem';
import {Logo} from '../Logo/Logo';
import {getFilteredHotels} from "../../selectors/getFilteredHotels";
import {connect} from "react-redux";
import Search from "../Search/Search";
import ModalWindow from "../ModalWindow";
import {Check} from "../SVG/Check";
import Success from "../Success/Success";

export class Header extends Component {
	constructor(props) {
		super(props);

		this.onModalOpen = this.onModalOpen.bind(this);
		this.onCloseModal = this.onCloseModal.bind(this);

		this.state = {
			modalIsOpen: false
		};
	}

	onModalOpen() {
		this.setState({modalIsOpen: true});
	}

	onCloseModal() {
		this.setState({modalIsOpen: false});
	}

	render() {
		return (
			<header className="header">
				<Logo/>
				<Search/>
				<button className="button is-success" onClick={this.onModalOpen}>Login<Check width={20} height={20} /></button>
				<ModalWindow isOpen={this.state.modalIsOpen} closeModal={this.onCloseModal} component={<LoginSystem/>}/>
			</header>
		)
	}
}

const mapStateToProps = (state: Object) => {
	return {
		hotels: getFilteredHotels(state.hotels, state.hotelsSearch)
	};
};

export default connect(mapStateToProps, null)(Header);
