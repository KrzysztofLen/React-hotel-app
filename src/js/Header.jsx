import React from 'react';
import LoginButton from "./LoginSystem/LoginButton";
import LogoutButton from './LoginSystem/LogoutButton';
import {Logo} from './Logo';

export default class Header extends React.Component {
	constructor(props) {
		super(props);

		this.state = {
			isLogged: false
		}
	}

	// console.log(startLogin);
	handleClick = () => {
		console.log('click');
		// this.setState({isLogged: true})
	};

	render() {
		return (
			<header className="header">
				<Logo/>
				<form action="#" className="search">
					<input type="text" className="search__input" placeholder="Search hotels"/>
				</form>
				<LogoutButton/>
				<LoginButton />
			</header>
		)
	}
};

