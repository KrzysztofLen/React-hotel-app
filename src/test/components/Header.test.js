import React from 'react';
import {shallow} from 'enzyme';
import Header from '../../../src/js/Header';

test('should render Header correctly', () => {
	const wrapper = shallow(<Header/>);
	expect(wrapper).toMatchSnapshot();
});