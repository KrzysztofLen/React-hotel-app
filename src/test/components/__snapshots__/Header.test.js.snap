// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          to="/"
>
          <img
                    alt="trillo logo"
                    className="logo"
                    src="logo.png"
          />
</NavLink>,
        <form
          action="#"
          className="search"
>
          <input
                    className="search__input"
                    placeholder="Search hotels"
                    type="text"
          />
</form>,
        <button
          className="btn btn--login"
>
          Login
</button>,
      ],
      "className": "header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "ariaCurrent": "true",
          "children": <img
            alt="trillo logo"
            className="logo"
            src="logo.png"
/>,
          "to": "/",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "trillo logo",
            "className": "logo",
            "src": "logo.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "action": "#",
          "children": <input
            className="search__input"
            placeholder="Search hotels"
            type="text"
/>,
          "className": "search",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "search__input",
            "placeholder": "Search hotels",
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        "type": "form",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Login",
          "className": "btn btn--login",
        },
        "ref": null,
        "rendered": "Login",
        "type": "button",
      },
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            to="/"
>
            <img
                        alt="trillo logo"
                        className="logo"
                        src="logo.png"
            />
</NavLink>,
          <form
            action="#"
            className="search"
>
            <input
                        className="search__input"
                        placeholder="Search hotels"
                        type="text"
            />
</form>,
          <button
            className="btn btn--login"
>
            Login
</button>,
        ],
        "className": "header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": <img
              alt="trillo logo"
              className="logo"
              src="logo.png"
/>,
            "to": "/",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "trillo logo",
              "className": "logo",
              "src": "logo.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "action": "#",
            "children": <input
              className="search__input"
              placeholder="Search hotels"
              type="text"
/>,
            "className": "search",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "search__input",
              "placeholder": "Search hotels",
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "type": "form",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Login",
            "className": "btn btn--login",
          },
          "ref": null,
          "rendered": "Login",
          "type": "button",
        },
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
